packages/flutter/lib/src/material/button.dart

```diff
@@ -451,35 +452,111 @@ class MaterialButton extends StatelessWidget {
 ///
 /// The primary purpose of this widget is to allow padding around [Material] widgets
 /// to trigger the child ink feature without increasing the size of the material.
-class _ButtonRedirectingHitDetectionWidget extends SingleChildRenderObjectWidget {
-  const _ButtonRedirectingHitDetectionWidget({
+class _ButtonPadding extends SingleChildRenderObjectWidget {
+  const _ButtonPadding({
     Key key,
     Widget child,
-    this.constraints
+    this.minHeight,
+    this.minWidth,
   }) : super(key: key, child: child);
 
-  final BoxConstraints constraints;
+  final double minWidth;
+  final double minHeight;
 
   @override
   RenderObject createRenderObject(BuildContext context) {
-    return new _RenderButtonRedirectingHitDetection(constraints);
+    return new _RenderButtonPadding(minWidth, minHeight);
   }
 
   @override
-  void updateRenderObject(BuildContext context, covariant _RenderButtonRedirectingHitDetection renderObject) {
-    renderObject.additionalConstraints = constraints;
+  void updateRenderObject(BuildContext context, covariant _RenderButtonPadding renderObject) {
+    renderObject
+      ..minHeight = minHeight
+      ..minWidth = minWidth;
   }
 }
 
-class _RenderButtonRedirectingHitDetection extends RenderConstrainedBox {
-  _RenderButtonRedirectingHitDetection (BoxConstraints additionalConstraints) : super(additionalConstraints: additionalConstraints);
+class _RenderButtonPadding extends RenderShiftedBox {
+  _RenderButtonPadding(this._minWidth, this._minHeight, [RenderBox child]) : super(child) ;
+
+  double get minWidth => _minWidth;
+  double _minWidth;
+  set minWidth(double value) {
+    if (_minWidth == value)
+      return;
+    _minWidth = value;
+    markNeedsLayout();
+  }
+
+  double get minHeight => _minHeight;
+  double _minHeight;
+  set minHeight(double value) {
+    if (_minHeight == value)
+      return;
+    _minHeight = value;
+    markNeedsLayout();
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    if (child != null)
+      return math.max(child.computeMinIntrinsicWidth(height), minWidth);
+    return 0.0;
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    if (child != null)
+      return math.max(child.computeMaxIntrinsicHeight(width), minHeight);
```

Comment:
If this is really supposed to be  `computeMaxIntrinsicHeight()`  (not min), maybe a comment explaining what's up.

Reply:
D'oh - I'm guessing this was the cause of several failures early on